spec.bs

```diff
@@ -8159,43 +8322,151 @@ To <dfn>batch or fetch trusted bidding signals</dfn> given a [=trusted bidding s
+  1. If |compressionGroupMap| does not [=map/contain=] |compressionGroupId|:
+    1. [=map/Set=] |compressionGroupMap|[|compressionGroupId|] to an empty [=map=], whose [=map/keys=] are integers
+      and [=map/values=] are [=maps=].
+  1. Let |partitionId| be 0.
+  1. If |group|'s [=interest group/execution mode=] is not "`group-by-origin`":
+    1. If |compressionGroupMap|[|compressionGroupId|] [=map/contains=] 0, set |partitionId| to
+      [=map/size=] of |compressionGroupMap|[|compressionGroupId|].
+    1. Otherwise, set |partitionId| to [=list/size=] of |compressionGroupMap|[|joiningOrigin|] plus 1.
+  1. [=map/Set=] |interestGroupIdMap|[|group|'s [=interest group/name=]] to [=tuple=] of
+    (|compressionGroupId|, |partitionId|).
+  1. If |compressionGroupMap|[|compressionGroupId|] does not [=map/contain=] |partitionId|, set
+    |compressionGroupMap|[|compressionGroupId|][|partitionId|] to a [=bidding partition=] with the following [=struct/items=]:
+    : [=bidding partition/id=]
+    :: |partitionId|
+
+    : [=bidding partition/namespace=]
+    :: The [=ordered map=] «[ "interestGroupNames" → « |group|'s [=interest group/name=] »,
+      "biddingKeys" → |group|'s [=interest group/trusted bidding signals keys=] ]»
+
+    : [=bidding partition/metadata=]
+    :: The [=ordered map=] «[ "experimentGroupId" → |experimentGroupId|, |slotSizeParams|[0] → |slotSizeParams|[1] ]»
   1. Otherwise:
-    1. [=Fetch the current outstanding trusted signals batch=] given |trustedBiddingSignalsBatcher|,
-      |signalsUrl|, |scriptOrigin|, |experimentGroupId|, |topLevelOrigin|, |slotSizeQueryParam|, and
-      |policyContainer|.
-    1. If |ig|'s [=interest group/trusted bidding signals keys=] is not null, set
-      |trustedBiddingSignalsBatcher|'s [=trusted bidding signals batcher/keys=] to a
-      [=list/clone=] of |ig|'s [=interest group/trusted bidding signals keys=].
-    1. Otherwise, set |trustedBiddingSignalsBatcher|'s [=trusted bidding signals batcher/keys=] to a
-       new [=list=].
-    1. Set |trustedBiddingSignalsBatcher|'s [=trusted bidding signals batcher/ig names=] to
-        « |igName| ».
-    1. Set |trustedBiddingSignalsBatcher|'s [=trusted bidding signals batcher/length limit=] to
-      |ig|'s [=interest group/max trusted bidding signals url length=].
+    1. [=list/Append=] |group|'s [=interest group/name=] into |compressionGroupMap|[|compressionGroupId|]
+      [|partitionId|]["interest_group_names"].
+    1. [=list/Append=] |group|'s [=interest group/trusted bidding signals keys=] into
+      |compressionGroupMap|[|compressionGroupId|][|partitionId|]["bidding_keys"].
+1. [=map/For each=] |id| → |group| in |compressionGroupMap|:
+  1. Let |compressionGroup| be an empty [=bidding compression group=].
+  1. Set |compressionGroup|'s [=bidding compression group/compression group id=] to |id|.
+  1. Set |compressionGroup|'s [=bidding compression group/partition=] to an empty [=list=].

```

Comment:
I had a comment about this which got resolved, but seems it's still not changed.
struct has no "empty" concept, no need to call them "empty"
```suggestion
  1. Let |compressionGroup| be a [=bidding compression group=], whose
    [=bidding compression group/compression group id=] is |id|, and
    [=bidding compression group/partition=] is an empty [=list=].
```

Reply:
Weird, now it is changed.